### ISSUE TITLE
[Feature] `Safe Mode` or `Crash Problem Finder`

### ISSUE THREAD

**Issue by sapioit** [BetterDiscord/BetterDiscord#1260]
**Is your feature request related to a problem? Please describe.**
I had a problem which made me manually disable mods 2-4 at a time, using the `plugins` settings page. Then the next day some plugins updated. Following that update, BetterDiscord kept crashing after a long loading time. It is not the first time when uploading a mod caused Discord to keep crashing, but most often the error was fixed by an update.

**Describe the feature you'd like**
`Crash Problem Finder`: When we are told BetterDiscord crashed and Discord loaded without BetterDiscord, please ad the option to select a mode in which we have the option to load plugins one at a time, pressing a button after each mod loads to load the next mod, while also showing us what mod will be loaded after pressing that button (maybe showing the file's name, in case reading the file causes the crash), so we can manually remove the file which causes the crash.

`Safe Mode`: Alternatively, there could be a safe-mode which to not load any theme or plugin, but still show us the enabled and disabled themes and plugins, so we can manually disable them and reload Discord until we find what keeps crashing it. It could be done by having us press or hold pressed a certain button or button combination while Discord is loading, to get into the safe mode.

**Additional context**
I'm surprised there is no such utility already implemented.

**Comment by Traktool** [comment]
Same problem here, BetterDiscord kept crashing again and again

I've found a temporary solution to disable all plugins
Do "Windows + R", type %appdata%/BetterDiscord/data/stable
Open plugin.json and set all values to "false"
Then save and relaunch Discord

**Comment by sapioit** [comment]
Thanks!

I think there might also be a need to allow the updating of the plugins, while in the safe-mode or crash-problem-finder-mode, so we can update the plugin which causes the crashing.

**Comment by Traktool** [comment]
> Thanks!
> 
> I think there might also be a need to allow the updating of the plugins, while in the safe-mode or crash-problem-finder-mode, so we can update the plugin which causes the crashing.

I've just find, 5 minutes ago, that "Unread Count Badges" make Discord crash, so if you have it, disable it and relaunch Discord with all your plugins enabled except it ;)

**Comment by sapioit** [comment]
Thank you very much! I was going to use the following system to find out the mod with problems, but you found the problem before I could. Steps of the system, which I'll leave here in case someone else has a similar problem in the future and with a different plugin:

1. Backup the `plugins.json` file.
2. Move half the plugins in a placeholder/temporary folder elsewhere.
3. Launch Discord.
4. If BetterDiscord crashed, repeat from Step 2, with the current plugins, and a different backup folder.
5. If BetterDiscord didn't crash and there is only plugin left in the folder, that is the plugin which crashed BetterDiscord, and should be disabled in a copy of the backed-up `plugins.json` which to replace the current version of `plugins.json` which got modified by BetterDiscord launching successfully, and move all the other plugins back into the folder from where you moved them to find the mod which crashed BetterDiscord.
6. If BetterDiscord didn't crash, move the current plugins in a different folder, and move back there the plugins moved after the last time BetterDiscord crashed, then repeat from Step 2.

**Comment by SmolAlli** [comment]
The talk of which plugins are crashing is not relevant to the discussion of the issue. Please join the [discord server](https://betterdiscord.app/invite) for support relating to crashing plugins.

**Comment by QbDesu** [comment]
I had a "Safe Mode" (Recovery Mode) similar to what you described in the works at some point as PR #1023
The code became quite messy and I ended up dropping the PR to start it from scratch when I get around to it.

Issue #1018 is an epic that describes various methods by which BD could identify faulty plugins, prevent crashes, and aid users in fixing problems on their own. It also describes a diagnostic mode that is similar to your description of a "Crash Problem Finder"

As to why nothing like that was implemented thus far.. it's easier said than done, it's more involved than it may seem. As for me I will not do another attempt at implementing a recovery mode until the context isolation changes from the `development` branch make their way to `main`.
